autometa/taxonomy/ncbi.py

```diff
@@ -92,332 +93,304 @@ def __str__(self):
-    parser = argparse.ArgumentParser(
-        description='Autometa NCBI utilities class')
-    parser.add_argument('ncbi', help='</path/to/ncbi/database/directory>')
-    parser.add_argument(
-        '--query-taxid',
-        help='query taxid to test NCBI class functionality',
-        default=1222,
-        type=int,
-    )
-    parser.add_argument('--verbose', help="add verbosity",
-                        action='store_true', default=True)
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
-        f'{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n'
-    )
-
-
-if __name__ == '__main__':
-    main()
+        lineage_a_taxids = {ancestor.get("taxid")
+                            for ancestor in self.lineage(taxid_A)}
+        lineage_b_taxids = {ancestor.get("taxid")
+                            for ancestor in self.lineage(taxid_B)}
+        common_ancestor = lineage_b_taxids.intersection(lineage_a_taxids)
+        common_ancestor.discard(1)  # This discards root
+        return True if common_ancestor else False
+
+    def is_valid_taxid(self, taxid):
+        """
+        Checks if the given `taxid` is a positive interger or not
+
+        Parameters
+        ----------
+        taxid : int
+            identifer for a taxon in NCBI taxonomy databases - `nodes.dmp`, `names.dmp` or `merged.dmp`
+
+        Returns
+        -------
+        taxid
+            `taxid` if the `taxid` is a positive integer
+
+        Raises
+        ------
+        ValueError
+            Provided `taxid` is not a positive integer
+        """
+        # This checks if an integer has been added as str, eg. "562"
+        if isinstance(taxid, str):
+            # float(taxid).is_integer() checks if it is something like 12.0 vs. 12.9.
+            if (taxid.isnumeric() or taxid.replace(".", "", 1).isnumeric()) and float(taxid).is_integer() and float(taxid) > 0:
+                # See https://stackoverflow.com/a/47764450/12671809
+                return int(float(taxid))
+            raise ValueError(
+                f"Taxid must be a positive integer! Given: {taxid}")
+        # `is_integer` returns error if only '12' is used, float(taxid) is needed
+        if not float(taxid).is_integer() or taxid < 0:
+            raise ValueError(
+                f"Taxid must be a positive integer! Given: {taxid}")
+        return int(taxid)
```

Comment:
This could be further simplified. Instead of multiple return statements, you can use a variable then return the variable if no issues are encountered.

- [Using](https://stackoverflow.com/a/49578517/13118765) the `string` lib to find digits, special characters, etc.

`import string`

```suggestion    
        if isinstance(taxid, str):
            invalid_chars = set(string.punctuation + string.ascii_letters)
            invalid_chars.discard('.')
            if {char for char in taxid if char in invalid_chars} or taxid.count('.') > 1:
                raise ValueError(f"taxid contains invalid character(s)! Given: {taxid}")            
            taxid = float(taxid)
        if isinstance(taxid, bool) or (isinstance(taxid, float) and not taxid.is_integer()):
            raise ValueError(f"taxid must be a positive integer! Given: {taxid}")
        taxid = int(taxid)
        if taxid <= 0:
            raise ValueError(f"Taxid must be a positive integer! Given: {taxid}")
        return taxid
```